src/components/shared/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from 'react';
+
+import NumberInputContainer from './NumberInput.container';
+import Fab from 'components/shared/Fab';
+
+/**
+ * This component is a
+ * Number Input.
+ *
+ * @author jeremy
+ *
+ * Created: January 11, 2019
+ **/
+
+interface Props {
+  value: number;
+  max: number;
+  min: number;
+  onChange: (value: number) => void;
+  step: number;
+
+  end?: string;
+  font?: string;
+  prefixColor?: string;
+  prefixSize?: string;
+  start?: string;
+  suffixColor?: string;
+  suffixSize?: string;
+  svgSize?: string;
+  textColor?: string;
+}
+
+interface State {
+  value: number;
+}
+
+class NumberInput extends React.Component<Props, State> {
+  static defaultProps: Props = {
+    value: 0,
+    max: Number.MAX_SAFE_INTEGER,
+    min: 0,
+    onChange: () => 0,
+    step: 1,
+  };
+
+  readonly state = {
+    value: this.props.value
+  };
+
+  render() {
+    const { value } = this.state;
+    const { max, min, prefixColor, suffixColor } = this.props;
+    const containerProps = {
+      ...this.props,
+      onChange: undefined // prevents Container typescript error
+    };
+    return (
+      <NumberInputContainer className="bee-number-input" {...containerProps}>
+        <Fab
+          clear
+          disabled={value === min}
+          height="24px"
+          icon="utils/minus-circle-outline"
+          iconColor={prefixColor || 'secondary'}
+          noPadding
+          onClick={this.handleMinus}
+          type="button"
+          width="24px"
+        />
+        <span>{this.state.value}</span>
```

Comment:
Should this wrap an [ `<input type='number'>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number) instead of just a span?

From the issue description, I see that there is an issue where users can clear the contents. Could we address that by wrapping an `input` here? (e.g. `onChange` check if text is empty/non-numeric and restore prior value when so)

Using `<input type='number'>` gives us:

* Accessibility (ability to tab in/out, semantics for screen readers)
* Flexibility (user can type a number instead of just clicking buttons)
* Numeric keyboards (on mobile)
* More stuff that I'm forgetting (probably)

Reply:
tried a few css tricks with `input`. Found out that it has a default `width` which makes it difficult to conditionally render css `padding` between svgs in the long run.

Talked with @kcvan pointed out that users don't actually need to tab into the number itself -- tabbing on airbnb just selects the buttons and they just use `div`s and `button`s.

![image](https://user-images.githubusercontent.com/18321302/51067557-2af73f80-15c8-11e9-98dd-a2a69809b9d3.png)